distributed/shuffle/_merge.py

```diff
@@ -243,15 +243,13 @@ def _cull_dependencies(
         all input partitions. This method does not require graph
         materialization.
         """
-        deps = defaultdict(set)
+        deps = defaultdict(frozenset)
         parts_out = parts_out or self._keys_to_parts(keys)
+        keys = {(self.name_input_left, i) for i in range(self.npartitions)}
+        keys |= {(self.name_input_right, i) for i in range(self.npartitions)}
+        keys = frozenset(keys)
```

Comment:
maybe add a comment why this is a frozenset. Using a frozenset is a bit uncommon and if we're honest this is just paranoia / testing